app/src/main/java/com/asfoundation/wallet/ui/iab/PaymentMethodsPresenter.kt

```diff
@@ -710,4 +798,57 @@ class PaymentMethodsPresenter(
     outState.putSerializable(FIAT_VALUE, cachedFiatValue)
     outState.putSerializable(PAYMENT_NAVIGATION_DATA, cachedPaymentNavigationData)
   }
+
+  private fun getPurchaseFiatValue(): Single<FiatValue> {
+    // TODO when adding new currency configurable on the settings we need to update this logic to be
+    //  aligned with the currency the user chooses
+    return interactor.getSkuDetails(paymentMethodsData.appPackage, paymentMethodsData.sku)
+      .map { product ->
+        val price = product.price
+        // NOTE we need to convert a double to a string in order to create the big decimal since the
+        // direct use of double won't result in the value you expect changing this will be a bad
+        // idea ate least for now
+        // Setup the appc value according with what we have on the microservices instead of
+        // converting, avoiding this way a additional call to the backend
+        transaction.amount(BigDecimal(price.appcoinsAmount.toString()))
+        transaction.productName = product.title
+        return@map FiatValue(
+          BigDecimal(price.amount.toString()),
+          price.currency,
+          price.currencySymbol
+        )
+      }.onErrorResumeNext(
+        if (getOriginalCurrency() == "APPC") {
+          interactor.convertAppcToLocalFiat(getOriginalValue().toDouble())
+        } else {
+          interactor.convertCurrencyToLocalFiat(
+            getOriginalValue().toDouble(),
+            getOriginalCurrency()
+          ).map { fiatValue ->
+            if (transaction.amount().compareTo(BigDecimal.ZERO) == 0) {
+              interactor.convertCurrencyToAppc(getOriginalValue().toDouble(), getOriginalCurrency())
+                .map { appcValue ->
+                  transaction.amount(appcValue.amount)
+                }
+            }
+            return@map fiatValue
+          }
+        })
+  }
```

Comment:
With the changes we discussed to TransactionBuilder this will become more obvious, but it will almost always be the case that if the original currency is not APPC, we will have to convert the currency to local fiat and the currency to APPC. The code makes it seem like there can be a case where this does not happen, but is there really a case?

If there is not, we could zip both calls and make it faster (convertCurrencyToLocalFiat + convertCurrencyToAppc).